src/Shell/Task/MigrationSnapshotTask.php

```diff
@@ -161,10 +161,65 @@ public function templateData()
     }
 
     /**
+     * Order tables based on foreign key dependencies so tables that are foreign keys
+     * to other are created first
+     *
+     * @param array $tables Tables to order.
+     * @return array Tables ordered.
+     */
+    public function orderTables($tables)
+    {
+        if (empty($tables)) {
+            return $tables;
+        }
+
+        $orderedTables = [];
+        foreach ($tables as $table) {
+            $tableSchema = $this->getCollection($this->connection)->describe($table);
+            $tableConstraints = $tableSchema->constraints();
+            if (isset($tableConstraints[0]) && $tableConstraints[0] === 'primary') {
+                unset($tableConstraints[0]);
+            }
+
+            if (!empty($tableConstraints)) {
+                foreach ($tableConstraints as $tableConstraint) {
+                    $constraint = $tableSchema->constraint($tableConstraint);
+
+                    if ($constraint['type'] !== 'foreign') {
+                        continue;
+                    }
+
+                    $refTable = $constraint['references'][0];
+
+                    $refIndex = array_search($refTable, $orderedTables);
+                    $tableIndex = array_search($table, $orderedTables);
+
+                    if ($refIndex === false && $tableIndex === false) {
+                        array_unshift($orderedTables, $refTable, $table);
+                    } elseif ($refIndex === false) {
+                        array_splice($orderedTables, $tableIndex, 0, $refTable);
+                    } elseif ($tableIndex === false) {
+                        array_splice($orderedTables, $refIndex+1, 0, $table);
+                    } elseif ($refIndex > $tableIndex) {
+                        unset($orderedTables[$refIndex]);
+                        array_splice($orderedTables, $tableIndex, 0, $refTable);
+                    }
+                }
+            }
```

Comment:
That might get tricky but I can give it a shot.
I'm afraid it will bring more overhead as it will need to reflect the same table multiple times.